dataland-internal-storage/src/main/kotlin/db/migration/V19__MigrateNoEvidenceFoundAndIncomplete.kt

```diff
@@ -0,0 +1,63 @@
+package db.migration
+
+import db.migration.utils.DataTableEntity
+import db.migration.utils.getOrJavaNull
+import db.migration.utils.migrateCompanyAssociatedDataOfDatatype
+import org.flywaydb.core.api.migration.BaseJavaMigration
+import org.flywaydb.core.api.migration.Context
+import org.json.JSONObject
+
+/**
+ * This migration script moves all SFDR Incomplete quality buckets to NoDataFound and removes the value NoEvidenceFound
+ * by putting NoDataFound into the quality bucket and setting the value to null
+ */
+class V19__MigrateNoEvidenceFoundAndIncomplete : BaseJavaMigration() {
+
+    /**
+     * Find all data points with a value of NoEvidenceFound and move it to the quality bucket or with quality Incomplete
+     * and move it to NoDataFound
+     */
+    private fun recursivelyChangeRelevantDataPointFields(
+        dataset: JSONObject,
+        objectName: String,
+    ) {
+        val obj = dataset.getOrJavaNull(objectName)
+        if (obj !== null && obj is JSONObject) {
+            if (obj.has("value") || obj.has("quality")) {
+                val value = obj.getOrJavaNull("value").toString()
+                val quality = obj.getOrJavaNull("quality").toString()
+                if (value == "NoEvidenceFound") {
+                    obj.put("value", JSONObject.NULL)
+                    obj.put("quality", "NoDataFound")
+                } else if (quality == "Incomplete") {
+                    obj.put("quality", "NoDataFound")
```

Comment:
Nur zum Verständnis:

D.h. wenn ein data point aktuell in production einen value hat, z.B. `revenue = 20.000.000`, und `quality = Incomplete`, ändern wir den quality Wert auf `NoDataFound`, weil wir damit ausdrücken wollen, dass für diesen value keine Quellenangabe gefunden werden konnte, um ihn zu bezeugen?

D.h. aus Business-Sicht wären die 20 Mio. geschätzt oder geraten (o.Ä.)

Reply:
wie besprochen